### PR TITLE
[risk=no] [RW-6073] Moving measurementconceptinfo client mapping to mapper

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
Moving measurementconceptinfo client mapper code from controller to mapper.

### CODE REVIEW

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
-    List<Concept> findSourceConcepts(@Param("conceptId") long conceptId,@Param("minCount") Integer minCount);
+    List<DbConcept> findSourceConcepts(@Param("conceptId") long conceptId,@Param("minCount") Integer minCount);
 
     @Query(value = "select c.* from concept c " +
             "join concept_relationship rel on rel.concept_id_2 = c.concept_id " +
             "and rel.concept_id_1 = :conceptId and rel.relationship_id = 'maps to' " +
             "where c.concept_id != :conceptId order by c.count_value desc",
             nativeQuery = true)
-    List<Concept> findConceptsMapsToParents(@Param("conceptId") long conceptId);
+    List<DbConcept> findConceptsMapsToParents(@Param("conceptId") long conceptId);
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Do you still need this method? It is no longer referenced.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
 
-    List<Concept> findByConceptName(String conceptName);
+    List<DbConcept> findByConceptName(String conceptName);
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Do you still need this method? It is no longer referenced.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
 
     @Query(value = "select c.* from concept c " +
             "where c.vocabulary_id in ('Gender', 'Race', 'Ethnicity')",
             nativeQuery = true)
-    List<Concept> findGenderRaceEthnicityFromConcept();
+    List<DbConcept> findGenderRaceEthnicityFromConcept();
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Do you still need this method? It is no longer referenced.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java [hunk 4]
```diff
@@ -66,41 +66,6 @@
             "order by c.vocabularyId\n")
     List<VocabularyCount> findVocabularyAllConceptCounts(String matchExp, String domainId);
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Do you still need this method? It is no longer referenced.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 5]
```diff
@@ -19,10 +23,22 @@
...
 public class ConceptService {
 
+    @PersistenceContext(unitName = "cdr")
+    private EntityManager entityManager;
+    @Autowired
+    private ConceptDao conceptDao;
+    @Autowired
+    private ConceptMapper conceptMapper;
+
     public static enum SearchType {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Modifier `static` is redundant for inner enums. Can be removed.
```suggestion
    public enum SearchType {
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 6]
```diff
@@ -19,10 +23,22 @@
...
         CONCEPT_SEARCH, SURVEY_COUNTS, DOMAIN_COUNTS;
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Unnecessary semicolon `;`
```suggestion
        CONCEPT_SEARCH, SURVEY_COUNTS, DOMAIN_COUNTS
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 7]
```diff
@@ -47,26 +63,21 @@ public ConceptIds(List<Long> standardConceptIds, List<Long> sourceConceptIds) {
 
     }
 
-    public enum StandardConceptFilter {
+    public enum StandardConceptFilterVal {
         ALL_CONCEPTS,
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: `ALL_CONCEPTS ` is never used.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 8]
```diff
@@ -145,16 +156,16 @@ public static String modifyMultipleMatchKeyword(String query, SearchType searchT
...
                         predicates.add(
                                 criteriaBuilder.or(
                                         criteriaBuilder.or(criteriaBuilder.isNull(root.get("standardConcept"))),
                                         criteriaBuilder.and(nonStandardConceptPredicates.toArray(new Predicate[0]))
                                 ));
 
-                    } else if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH)) {
+                    } else if (standardConceptFilter.equals(StandardConceptFilterVal.STANDARD_OR_CODE_ID_MATCH)) {
                         if (keyword != null) {
                             if(keyword != null){
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: No need to wrap the same condition in itself
```suggestion
                        if (keyword != null) {
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 9]
```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
...
+        }
+
+        String domainId = null;
+        if (searchConceptsRequest.getDomain() != null) {
+            domainId = CommonStorageEnums.domainToDomainId(searchConceptsRequest.getDomain());
+        }
+
+        StandardConceptFilterVal convertedConceptFilter = StandardConceptFilterVal.valueOf(standardConceptFilter.name());
+
+        Slice<DbConcept> concepts = null;
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Variable 'concepts' initializer 'null' is redundant 
```suggestion
        Slice<DbConcept> concepts;
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 10]
```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
...
+            }
+        }
+
+        if(response.getMatchType() == null && response.getStandardConcepts() == null){
+            response.setMatchType(MatchType.NAME);
+        }
+
+        List<Concept> conceptList = new ArrayList<>();
+
+        if (concepts != null) {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Variable `concepts` will always be true. Need to check context.
```suggestion
        if (concepts.getContent() != null) {
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 11]
```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
...
+        }
+        List<Long> drugConcepts = new ArrayList<>();
+
+        if(searchConceptsRequest.getDomain() != null && searchConceptsRequest.getDomain().equals(Domain.DRUG) && searchConceptsRequest.getQuery() != null && !searchConceptsRequest.getQuery().isEmpty()) {
+            List<Long> drugMatchedConcepts = new ArrayList<>();
+            drugMatchedConcepts = getDrugIngredientsByBrand(searchConceptsRequest.getQuery());
+            if(drugMatchedConcepts.size() > 0) {
+                drugConcepts = drugMatchedConcepts;
+            }
+        }
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: `getDrugIngredientsByBrand` will return an empty list. No need to use 2 list.
```suggestion
        List<Long> drugConcepts = new ArrayList<>();

        if(searchConceptsRequest.getDomain() != null && searchConceptsRequest.getDomain().equals(Domain.DRUG) && searchConceptsRequest.getQuery() != null && !searchConceptsRequest.getQuery().isEmpty()) {
            drugConcepts = getDrugIngredientsByBrand(searchConceptsRequest.getQuery());
        }
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 12]
```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
...
+
+        Integer minCount = searchConceptsRequest.getMinCount();
+        if(minCount == null){
+            minCount = 1;
+        }
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Can use optional
```suggestion
        Integer minCount = Optional.ofNullable(searchConceptsRequest.getMinCount()).orElse(1);
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 13]
```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
...
+
+        StandardConceptFilter standardConceptFilter = searchConceptsRequest.getStandardConceptFilter();
+
+
+        if(searchConceptsRequest.getQuery() == null || searchConceptsRequest.getQuery().isEmpty()){
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: `org.apache.commons.lang3.StringUtils.isEmpty()` checks for empty ("") or null
```suggestion
       if(StringUtils.isEmpty(searchConceptsRequest.getQuery())){
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java [hunk 14]
```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
...
+
+        List<Concept> conceptList = new ArrayList<>();
+
+        if (concepts != null) {
+            conceptList = concepts.getContent().stream()
+                    .map(conceptMapper::dbModelToClient)
+                    .collect(Collectors.toList());
+
+            if(response.getStandardConcepts() != null) {
+                conceptList = conceptList.stream().filter(c -> Long.valueOf(c.getConceptId()) != Long.valueOf(response.getSourceOfStandardConcepts())).collect(Collectors.toList());
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]:
```suggestion
                conceptList = conceptList.stream().filter(c -> !c.getConceptId().equals(response.getSourceOfStandardConcepts())).collect(Collectors.toList());
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 15]
```diff
@@ -0,0 +1,366 @@
...
+    private int canSelect;
+    private int hasCounts;
+    private DbMeasurementConceptInfo dbMeasurementConceptInfo = null;
+    private String graphToShow;
+
+    public DbConcept() {
+    }
+
+    // Copy constructor for copying everything but synonyms
+    public DbConcept(DbConcept a) {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:47]: Constructor is never used.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 16]
```diff
@@ -0,0 +1,366 @@
...
+        this.hasCounts = hasCounts;
+    }
+
+    public DbConcept hasCounts(int hasCounts) {
+        this.hasCounts = hasCounts;
+        return this;
+    }
+
+    @Transient
+    public String getGraphToShow() {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:284]: This method is no longer referenced in codebase.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 17]
```diff
@@ -0,0 +1,366 @@
...
+        if (getDomainId().equals("Measurement") && getDbMeasurementConceptInfo() != null && getDbMeasurementConceptInfo().getHasValues() == 1) {
+            return "Values";
+        } else {
+            return "Sex Assigned at Birth";
+        }
+    }
+    public void setGraphToShow() {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:291]: This method is no longer referenced in codebase.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 18]
```diff
@@ -0,0 +1,366 @@
...
+        if (getDomainId().equals("Measurement") && getDbMeasurementConceptInfo() != null && getDbMeasurementConceptInfo().getHasValues() == 1) {
+            this.graphToShow = "Values";
+        } else {
+            this.graphToShow = "Sex Assigned at Birth";
+        }
+    }
+    public DbConcept graphToShow(String graphToShow) {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:298]: This method is no longer referenced in codebase.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 19]
```diff
@@ -0,0 +1,366 @@
...
+    public List<String> getSynonyms() {
+        return synonyms;
+    }
+    public void setSynonyms(List<String> synonyms) {
+        this.synonyms = synonyms;
+    }
+    public DbConcept synonyms(List<String> synonyms) {
+        this.synonyms = synonyms;
+        return this;
+    }
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:313]: These methods are no longer referenced in codebase.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 20]
```diff
@@ -0,0 +1,366 @@
...
+
+    @Transient
+    public List<String> getDrugBrands() { return drugBrands; }
+    public void setDrugBrands(List<String> drugBrands) { this.drugBrands = drugBrands; }
+    public DbConcept drugBrands(List<String> drugBrands) {
+        this.drugBrands = drugBrands;
+        return this;
+    }
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:321]: These methods are no longer referenced in codebase.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java [hunk 21]
```diff
@@ -0,0 +1,366 @@
...
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name="concept_id", insertable=false, updatable=false)
+    public DbMeasurementConceptInfo getDbMeasurementConceptInfo() {
+        return dbMeasurementConceptInfo;
+    }
+    public void setDbMeasurementConceptInfo(DbMeasurementConceptInfo dbMeasurementConceptInfo) {
+        this.dbMeasurementConceptInfo = dbMeasurementConceptInfo;
+    }
+
+    public DbConcept dbMeasurementConceptInfo(DbMeasurementConceptInfo dbMeasurementConceptInfo) {
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:332]: This method is no longer referenced in codebase.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbMeasurementConceptInfo.java [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+
+@Entity
+@Table(name = "measurement_concept_info")
+public class DbMeasurementConceptInfo {
+    private Long conceptId;
+    private int hasValues;
+
+    @OneToOne(fetch=FetchType.LAZY)
+    @JoinColumn(name="concept_id")
+    private Concept concept;
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/DbMeasurementConceptInfo.java:22]: This variable is never used. Please remove.

File: public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java [hunk 23]
```diff
@@ -736,4 +737,44 @@ public float normalizeBinWidth(Float bMin, Float bMax) {
...
+        Multimap<String, AchillesResult> ageAnalysisResultsByQuestion = Multimaps.index(
+                analysisMap.get(CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.SURVEY_AGE_ANALYSIS_ID)).getResults(), AchillesResult::getStratum2);
+        Multimap<String, AchillesResult> versionAnalysisResultsByQuestion = Multimaps.index(
+                analysisMap.get(CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.SURVEY_VERSION_ANALYSIS_ID)).getResults(), AchillesResult::getStratum2);
+
+        for(SurveyMetadata q: questions) {
+            Analysis countAnalysis = analysisMap.get(CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.SURVEY_COUNT_ANALYSIS_ID));
+            if (countAnalysis != null) {
+                Analysis ca = achillesMapper.makeCopyAnalysis(countAnalysis);
+                ca.setResults(countAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId())).stream().collect(Collectors.toList()));
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java]:
```suggestion
                ca.setResults(new ArrayList<>(
                    countAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId()))));
```

File: public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java [hunk 24]
```diff
@@ -736,4 +737,44 @@ public float normalizeBinWidth(Float bMin, Float bMax) {
...
+                q.setCountAnalysis(ca);
+            }
+            Analysis genderAnalysis = analysisMap.get(CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.SURVEY_GENDER_ANALYSIS_ID));
+            if (genderAnalysis != null) {
+                Analysis ga = achillesMapper.makeCopyAnalysis(genderAnalysis);
+                ga.setResults(genderAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId())).stream().collect(Collectors.toList()));
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java]:
```suggestion
                ga.setResults(new ArrayList<>(
                    genderAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId()))));
```

File: public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java [hunk 25]
```diff
@@ -736,4 +737,44 @@ public float normalizeBinWidth(Float bMin, Float bMax) {
...
+                q.setGenderAnalysis(ga);
+            }
+            Analysis ageAnalysis = analysisMap.get(CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.SURVEY_AGE_ANALYSIS_ID));
+            if (ageAnalysis != null) {
+                Analysis aa = achillesMapper.makeCopyAnalysis(ageAnalysis);
+                aa.setResults(ageAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId())).stream().collect(Collectors.toList()));
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java]:
```suggestion
                aa.setResults(new ArrayList<>(
                    ageAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId()))));
```

File: public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java [hunk 26]
```diff
@@ -736,4 +737,44 @@ public float normalizeBinWidth(Float bMin, Float bMax) {
...
+                q.setAgeAnalysis(aa);
+            }
+            Analysis versionAnalysis = analysisMap.get(CommonStorageEnums.analysisIdFromName(AnalysisIdConstant.SURVEY_VERSION_ANALYSIS_ID));
+            if (versionAnalysis != null) {
+                Analysis aa = achillesMapper.makeCopyAnalysis(versionAnalysis);
+                aa.setResults(versionAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId())).stream().collect(Collectors.toList()));
```
Comment (freemabd) [inline public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java]:
```suggestion
                aa.setResults(new ArrayList<>(
                    versionAnalysisResultsByQuestion.get(String.valueOf(q.getConceptId()))));
```

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java [hunk 27]
```diff
@@ -24,17 +24,16 @@
...
     ConceptRelationshipDao conceptRelationshipDao;
 
 
-    Concept concept1 = makeConcept(123L, "concept a", "S", "conceptA", "classId", "V1", "Condition", 123L, 0.2F, 0L, 1);
-    Concept concept2 = makeConcept(456L, "concept b", null, "conceptB", "classId", "V2", "Condition", 456L, 0.2F, 0L, 1);
-    Concept concept3 = makeConcept(789L, "concept c", null, "conceptC", "classId", "V3", "Condition", 567L, 0.2F, 0L, 1);
-    Concept concept4 = makeConcept(1234L, "concept D test concept 1", null, "conceptD", "classId", "V4", "Measurement", 567L, 0.2F, 0L, 1);
-    Concept concept5 = makeConcept(5678L, "concept E test concept 1", "S", "conceptE", "classId", "V5", "Condition", 1234L, 0.2F, 0L, 1);
-    Concept concept6 = makeConcept(7890L, "concept F test concept 2", null, "conceptF", "classId", "V6", "Condition", 1234L, 0.2f, 0L, 1);
+    private DbConcept concept1 = makeConcept(123L, "concept a", "S", "conceptA", "classId", "V1", "Condition", 123L, 0.2F, 0L, 1);
```
Comment (freemabd) [inline public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]: nit: make final
```suggestion
    private final DbConcept concept1 = makeConcept(123L, "concept a", "S", "conceptA", "classId", "V1", "Condition", 123L, 0.2F, 0L, 1);
```

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java [hunk 28]
```diff
@@ -24,17 +24,16 @@
...
+    private DbConcept concept2 = makeConcept(456L, "concept b", null, "conceptB", "classId", "V2", "Condition", 456L, 0.2F, 0L, 1);
```
Comment (freemabd) [inline public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]:
```suggestion
    private final DbConcept concept2 = makeConcept(456L, "concept b", null, "conceptB", "classId", "V2", "Condition", 456L, 0.2F, 0L, 1);
```

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java [hunk 29]
```diff
@@ -24,17 +24,16 @@
...
+    private DbConcept concept3 = makeConcept(789L, "concept c", null, "conceptC", "classId", "V3", "Condition", 567L, 0.2F, 0L, 1);
```
Comment (freemabd) [inline public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]:
```suggestion
    private final DbConcept concept3 = makeConcept(789L, "concept c", null, "conceptC", "classId", "V3", "Condition", 567L, 0.2F, 0L, 1);
```

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java [hunk 30]
```diff
@@ -24,17 +24,16 @@
...
+    private DbConcept concept4 = makeConcept(1234L, "concept D test concept 1", null, "conceptD", "classId", "V4", "Measurement", 567L, 0.2F, 0L, 1);
```
Comment (freemabd) [inline public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]:
```suggestion
    private final DbConcept concept4 = makeConcept(1234L, "concept D test concept 1", null, "conceptD", "classId", "V4", "Measurement", 567L, 0.2F, 0L, 1);
```

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java [hunk 31]
```diff
@@ -24,17 +24,16 @@
...
+    private DbConcept concept5 = makeConcept(5678L, "concept E test concept 1", "S", "conceptE", "classId", "V5", "Condition", 1234L, 0.2F, 0L, 1);
```
Comment (freemabd) [inline public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]:
```suggestion
    private final DbConcept concept5 = makeConcept(5678L, "concept E test concept 1", "S", "conceptE", "classId", "V5", "Condition", 1234L, 0.2F, 0L, 1);
```

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java [hunk 32]
```diff
@@ -24,17 +24,16 @@
...
+    private DbConcept concept6 = makeConcept(7890L, "concept F test concept 2", null, "conceptF", "classId", "V6", "Condition", 1234L, 0.2f, 0L, 1);
```
Comment (freemabd) [inline public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]:
```suggestion
    private final DbConcept concept6 = makeConcept(7890L, "concept F test concept 2", null, "conceptF", "classId", "V6", "Condition", 1234L, 0.2f, 0L, 1);
```

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java (hunk 19 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:313]: The synonyms are used in this component view in ui 
https://github.com/all-of-us/data-browser/blob/master/public-ui/src/app/data-browser/views/ehr-view/ehr-view.component.ts#L410

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java (hunk 20 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:321]: Also, the drugbrands https://github.com/all-of-us/data-browser/blob/master/public-ui/src/app/data-browser/views/ehr-view/ehr-view.component.ts#L411

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbMeasurementConceptInfo.java (hunk 22 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/model/DbMeasurementConceptInfo.java:22]: This join relationship is used here https://github.com/all-of-us/data-browser/blob/master/public-ui/src/app/data-browser/views/ehr-view/ehr-view.component.ts#L311 to filter lab values vs orders.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java (hunk 21 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:332]: This join relationship is used here https://github.com/all-of-us/data-browser/blob/master/public-ui/src/app/data-browser/views/ehr-view/ehr-view.component.ts#L311 to filter lab values vs orders.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Removed thanks.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Removed thanks.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Removed thanks.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java (hunk 4 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Removed thanks.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 6 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 7 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Removed thanks.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 8 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 10 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 9 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 11 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 12 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 13 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 14 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Done thanks.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java (hunk 15 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:47]: Removed.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java (hunk 17 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/model/DbConcept.java:291]: Removed these.

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java (hunk 28 above)
Comment (SrushtiGangireddy) [reply public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]: Done thanks

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java (hunk 5 above)
Comment (SrushtiGangireddy) [reply public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java]: Changed this.

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java (hunk 30 above)
Comment (SrushtiGangireddy) [reply public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]: Changed this.

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java (hunk 29 above)
Comment (SrushtiGangireddy) [reply public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]: Changed thanks

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java (hunk 32 above)
Comment (SrushtiGangireddy) [reply public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]: Changed thanks

File: public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java (hunk 31 above)
Comment (SrushtiGangireddy) [reply public-api/src/test/java/org/pmiops/workbench/cdr/dao/ConceptRelationshipDaoTest.java]: Changed thanks

**Pull request closed without merging**